packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts

```diff
@@ -82,25 +98,100 @@ export class EditorSettingsSidebarComponent {
 	 * @returns {Promise<void>} No return value.
 	 */
 	async clickTab( tabName: EditorSidebarTab ): Promise< void > {
-		await this.frame.click( selectors.tabButton( tabName ) );
-		await this.frame.waitForSelector( selectors.activeTabButton( tabName ) );
+		const locator = this.frameLocator.locator( selectors.tabButton( tabName ) );
+		await locator.click();
+
+		const classAttributes = await locator.getAttribute( 'class' );
+		if ( ! classAttributes?.includes( 'is-active' ) ) {
+			throw new Error( `Failed to verify ${ tabName } is active.` );
+		}
 	}
 
 	/**
-	 * If the provided sidebar section is collapsed, toggles it to be expanded.
+	 * Expands a collapsed section of the sidebar.
 	 *
-	 * @param {EditorSidebarSection} sectionName Name of section.
+	 * If the section is already open, this method will pass.
+	 *
+	 * @param {ArticleSections} name Name of section to be expanded.
 	 * @returns {Promise<void>} No return value.
 	 */
-	async expandSection( sectionName: EditorSidebarSection ): Promise< void > {
-		// Avoid the wpcalypso/staging banner
+	async expandSection( name: ArticleSections ): Promise< void > {
+		if ( await this.targetIsOpen( selectors.section( name ) ) ) {
+			return;
+		}
+
+		// Avoid the wpcalypso/staging banner.
 		await this.scrollToBottomOfSidebar();
-		if ( ! ( await this.frame.isVisible( selectors.expandedSection( sectionName ) ) ) ) {
-			await this.frame.click( selectors.sectionToggle( sectionName ) );
+
+		const sectionLocator = this.frameLocator.locator( selectors.section( name ) );
+		await sectionLocator.click();
+
+		const newState = await this.targetIsOpen( selectors.section( name ) );
+		if ( ! newState ) {
```

Comment:
Tiny optional style nit: but with boolean variables, how do you feel about variable names that declare something that could be true or false? That is, something like `sectionIsOpen` instead of `newState`. I personally think that makes it easier to read the code flow! But, that's just my personal take, and definitely not required.

Reply:
Got it, replacing it with variable `expanded` which is a lot more intuitive.